src/core/server/uuid/manage_uuid.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import uuid from 'uuid';
+import { promisify } from 'util';
+import * as fs from 'fs';
+import { join } from 'path';
+import { take } from 'rxjs/operators';
+import { IConfigService } from '../config';
+import { PathConfigType } from '../path';
+import { HttpConfigType } from '../http';
+
+const FILE_ENCODING = 'utf8';
+const FILE_NAME = 'uuid';
+
+const readFile = promisify(fs.readFile);
+const writeFile = promisify(fs.writeFile);
+
+export async function manageInstanceUuid(configService: IConfigService): Promise<string> {
+  const [pathConfig, serverConfig] = await Promise.all([
+    configService
+      .atPath<PathConfigType>('path')
+      .pipe(take(1))
+      .toPromise(),
+    configService
+      .atPath<HttpConfigType>('server')
+      .pipe(take(1))
+      .toPromise(),
+  ]);
+
+  const uuidFilePath = join(pathConfig.data, FILE_NAME);
+
+  const uuidFromFile = await readUUIDFromFile(uuidFilePath);
```

Comment:
super nit: can we capitalize all these UUID names with the same convention used elsewhere: `readUuidFromFile`

Reply:
Thanks. I'm more of a `UUID` person myself, so I had to rename all theses after asking about our conventions. Missed some, will fix that.